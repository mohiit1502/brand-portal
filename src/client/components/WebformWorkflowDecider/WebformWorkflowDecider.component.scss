.c-WebformWorkflowDecider {
    .page-title{
        font-weight: bold;
        font-size: 2rem;
        padding: 1rem 9rem;
        //background: linear-gradient(to bottom, #8cd3f7, white);

    }

    .page-header {
        font-weight: bold;
        padding-bottom: 0.5rem;
    }

    .page-body {
        padding: 1rem 11rem;
    }
    .message-content {
        font-size: 20px;
    }

    @media only screen and (min-width: 1800px) {
        .page-title{
            font-weight: bold;
            font-size: 2rem;
            padding: 1rem 8rem;
            padding-top: 4rem;
        } 
        .page-body {
            padding: 1rem 11rem;
        }
    }
}
