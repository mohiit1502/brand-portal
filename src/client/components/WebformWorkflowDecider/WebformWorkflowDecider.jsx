/* eslint-disable filenames/match-regex */
import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {dispatchMetadata} from "../../actions/content/content-actions";
import {dispatchWebformState} from "../../actions/webform/webform-action";
import { WebformCta, WebformLandingPage } from "..";
import HomeHeader from "../custom-components/headers/home-header";
import Footer from "../Footer";
import Webform from "../Webform/Webform";
import FORMFIELDCONFIG from "../../config/formsConfig/form-field-meta";
import WEBFORMCONFIG from "../../config/contentDescriptors/webform";
import ContentRenderer from "../../utility/ContentRenderer";
import "./WebformWorkflowDecider.component.scss";
import Http from "../../utility/Http";
import CONSTANTS from "../../constants/constants";
import mixpanel from "../../utility/mixpanelutils";

class WebformWorkflowDecider extends React.Component {
  constructor(props) {
    super(props);
    const contentRenderer = new ContentRenderer();
    this.getContent = contentRenderer.getContent.bind(this);
    this.commonClickHandler = this.commonClickHandler.bind(this);
    this.dispatchWebformState = this.props.dispatchWebformState;
    this.state = {
      webformConfig: WEBFORMCONFIG
    };
  }

  componentDidMount() {
    try {
      this.setState({webformConfig: WEBFORMCONFIG});
      Http.get("/api/webformConfig")
        .then(response => {
          if (response.body) {
            try {
              response = JSON.parse(response.body);
              this.setState({webformConfig: response});
            } catch (e) {
              console.log(e);
            }
          }
        });

        this.props.dispatchMetadata(FORMFIELDCONFIG);
        Http.get("/api/formConfig")
        .then(response => {
          if (response.body) {
            try {
              response = JSON.parse(response.body);
              //response = FORMFIELDCONFIG;
              this.props.dispatchMetadata(response);
            } catch (e) {
                console.log(e);
            }
          }
        });
        if (!mixpanel.getToken()) {
          Http.get("/api/mixpanelConfig")
          .then(res => {
            mixpanel.intializeMixpanel(res.body.projectToken, res.body.enableTracking);
          }).catch(e => mixpanel.intializeMixpanel(CONSTANTS.MIXPANEL.PROJECT_TOKEN));
        }

      } catch (err) {
        console.log(err);
        this.props.dispatchMetadata(FORMFIELDCONFIG);
        this.setState({webformConfig: WEBFORMCONFIG});
    }
  }

  commonClickHandler(e) {
    console.log(this.props.webformWorkflow);
    this.dispatchWebformState(e.target.value);
  }

  render() {
      let configuration, childComponent;
      if (this.state.webformConfig && this.props.webformWorkflow === CONSTANTS.WEBFORM.CTA) {
        configuration = this.state.webformConfig.ctaPageConfig;
        childComponent = (<WebformCta configuration= {configuration} getContent={this.getContent}/>);

      } else if (this.state.webformConfig && this.props.webformWorkflow === CONSTANTS.WEBFORM.CLAIM_SUBMISSION) {
        configuration = this.state.webformConfig.webform;
        childComponent = this.props.webformFieldsConfiguration
          ? <Webform
              getContent={this.getContent}
              webformFieldsConfiguration={this.props.webformFieldsConfiguration}
              configuration={configuration}
              dispatchWebformState={this.dispatchWebformState}/>
          : null;
      }

      return (
        <div className="c-WebformWorkflowDecider">
          <HomeHeader isWebform={true} hideHelp={this.props.webformWorkflow !== CONSTANTS.WEBFORM.CLAIM_SUBMISSION} config={this.state.webformConfig ? this.state.webformConfig.webform : {}} />
          <div className="">
            {childComponent}
          </div>
          <div className="fixed-bottom">
            <Footer isWebform={true}/>
          </div>
        </div>
      );
  }
}

WebformWorkflowDecider.propTypes = {
  dispatchMetadata: PropTypes.func,
  dispatchWebformState: PropTypes.func,
  webformWorkflow: PropTypes.string,
  webformFieldsConfiguration: PropTypes.object
};

const mapDispatchToProps = {
  dispatchMetadata,
  dispatchWebformState
};

const mapStateToProps = state => {
  return {
    webformWorkflow: state.webform.state,
    webformFieldsConfiguration: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.WEBFORM
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(WebformWorkflowDecider);
