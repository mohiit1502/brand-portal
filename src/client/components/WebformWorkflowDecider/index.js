import WebformWorkflowDecider from './WebformWorkflowDecider.jsx';

export default WebformWorkflowDecider;
