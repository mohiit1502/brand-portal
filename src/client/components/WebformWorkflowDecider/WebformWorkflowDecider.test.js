import React from 'react';
import WebformWorkflowDecider from './WebformWorkflowDecider';

describe('WebformWorkflowDecider', () => {
   it('renders without error', () => {

   });
});
