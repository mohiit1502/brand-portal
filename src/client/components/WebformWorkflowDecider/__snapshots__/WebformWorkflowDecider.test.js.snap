// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebformWorkflowDecider renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockNextContext
    pathname="/help"
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/help",
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/help",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(WebformWorkflowDecider)
          dispatchWebformState={[Function]}
          webformWorkflow=""
        >
          <WebformWorkflowDecider
            dispatchWebformState={[Function]}
            webformWorkflow={Object {}}
          >
            <div
              className="c-WebformWorkflowDecider"
            >
              <Connect(HomeHeader)
                config={
                  Object {
                    "disclaimer": Object {
                      "actionBtnText": "OK",
                      "btnText": "View disclaimer",
                      "classes": "d-inline-block anchor-styled mb-35rem",
                      "modalDisclaimerText": "To make a claim of IP infringement, you will be required to declare, under penalty of perjury, that the information provided is correct and that you are the IP rights owner or an authorized agent. If the information you submit is inaccurate or incomplete, we may be unable to process or respond to your request. If you have questions about IP or whether your rights have been infringed, consider seeking legal advice.",
                      "modalHeaderText": "Disclaimer",
                    },
                    "header": Object {
                      "classes": "h4 font-weight-bold",
                      "text": "Intellectual Property (IP) Claim Form",
                    },
                    "help": Object {
                      "actionBtnText": "OK",
                      "classes": "d-inline-block anchor-styled",
                      "modalHeaderText": "Help",
                      "modalHelpText": Object {
                        "content": Object {
                          "partial1": Object {
                            "anchor1": Object {
                              "href": "mailto:ipinvest@walmart.com",
                              "text": "ipinvest@walmart.com",
                            },
                            "chunk1": "If you have any issues with filing a claim, please reach out to ",
                            "chunk2": " for further assistance.",
                          },
                        },
                      },
                    },
                    "subText": Object {
                      "classes": "d-inline-block mr-3 mb-35rem",
                      "text": "All of the following fields are required",
                    },
                    "titleClass": "ml-8",
                  }
                }
                hideHelp={true}
                isWebform={true}
              >
                <HomeHeader
                  config={
                    Object {
                      "disclaimer": Object {
                        "actionBtnText": "OK",
                        "btnText": "View disclaimer",
                        "classes": "d-inline-block anchor-styled mb-35rem",
                        "modalDisclaimerText": "To make a claim of IP infringement, you will be required to declare, under penalty of perjury, that the information provided is correct and that you are the IP rights owner or an authorized agent. If the information you submit is inaccurate or incomplete, we may be unable to process or respond to your request. If you have questions about IP or whether your rights have been infringed, consider seeking legal advice.",
                        "modalHeaderText": "Disclaimer",
                      },
                      "header": Object {
                        "classes": "h4 font-weight-bold",
                        "text": "Intellectual Property (IP) Claim Form",
                      },
                      "help": Object {
                        "actionBtnText": "OK",
                        "classes": "d-inline-block anchor-styled",
                        "modalHeaderText": "Help",
                        "modalHelpText": Object {
                          "content": Object {
                            "partial1": Object {
                              "anchor1": Object {
                                "href": "mailto:ipinvest@walmart.com",
                                "text": "ipinvest@walmart.com",
                              },
                              "chunk1": "If you have any issues with filing a claim, please reach out to ",
                              "chunk2": " for further assistance.",
                            },
                          },
                        },
                      },
                      "subText": Object {
                        "classes": "d-inline-block mr-3 mb-35rem",
                        "text": "All of the following fields are required",
                      },
                      "titleClass": "ml-8",
                    }
                  }
                  hideHelp={true}
                  isWebform={true}
                  toggleModal={[Function]}
                >
                  <nav
                    className="navbar navbar-expand-md navbar-dark home-header-nav home-header-nav-z-index"
                  >
                    <Link
                      className="navbar-brand walmart-brand"
                      to="/dashboard"
                    >
                      <LinkAnchor
                        className="navbar-brand walmart-brand"
                        href="/dashboard"
                        navigate={[Function]}
                      >
                        <a
                          className="navbar-brand walmart-brand"
                          href="/dashboard"
                          onClick={[Function]}
                        >
                          <img
                            src="WMWhite-horizontal.svg"
                          />
                        </a>
                      </LinkAnchor>
                    </Link>
                    <button
                      aria-controls="collapsible-header"
                      aria-expanded="false"
                      aria-label="Toggle navigation"
                      className="navbar-toggler"
                      data-target="#collapsible-header"
                      data-toggle="collapse"
                      type="button"
                    >
                      <span
                        className="navbar-toggler-icon"
                      />
                    </button>
                    <div
                      className="collapse navbar-collapse navbar-collapsible-header"
                      id="collapsible-header"
                    >
                      <ul
                        className="navbar-nav ml-auto"
                      >
                        <li
                          className="nav-item nav-item-help mx-4"
                        />
                      </ul>
                    </div>
                  </nav>
                </HomeHeader>
              </Connect(HomeHeader)>
              <div
                className=""
              />
              <div
                className="fixed-bottom"
              >
                <Footer
                  isWebform={true}
                >
                  <div
                    className="c-Footer"
                  >
                    <div
                      className="row"
                    >
                      <div
                        className="col"
                      >
                        <div
                          className="c-Footer__copyrightContainer"
                        >
                          <span
                            className="c-Footer__copyrightText float-left"
                          >
                            Â© 2021 Walmart. All rights reserved.
                            <a
                              className="c-Footer__privacyNav float-right"
                              href="https://corporate.walmart.com/privacy-security"
                              target="_blank"
                            >
                              Privacy Policy
                            </a>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </Footer>
              </div>
            </div>
          </WebformWorkflowDecider>
        </Connect(WebformWorkflowDecider)>
      </Router>
    </MemoryRouter>
  </MockNextContext>
</Provider>
`;
